### PR TITLE
feat(dws): add new resource to bind public domain name for cluster

### PR DESCRIPTION
**Pull request by wuzhuanhong**
**What this PR does / why we need it**:

Add new resource public domain name associate with DWS cluster.
Add related document and acceptance test.

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note

```

## PR Checklist

* [x] Tests added/passed.

```
 make testacc TEST=./huaweicloud/services/acceptance/dws TESTARGS='-run TestAccResourceCluster_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/dws -v -run TestAccResourceCluster_ -timeout 360m -parallel 4
=== RUN   TestAccResourceCluster_basicV1
=== PAUSE TestAccResourceCluster_basicV1
=== RUN   TestAccResourceCluster_basicV2
=== PAUSE TestAccResourceCluster_basicV2
=== RUN   TestAccResourceCluster_basicV2_mutilAZs
=== PAUSE TestAccResourceCluster_basicV2_mutilAZs
=== CONT  TestAccResourceCluster_basicV1
=== CONT  TestAccResourceCluster_basicV2_mutilAZs
=== CONT  TestAccResourceCluster_basicV2
--- PASS: TestAccResourceCluster_basicV2_mutilAZs (1685.17s)
--- PASS: TestAccResourceCluster_basicV1 (2829.21s)
--- PASS: TestAccResourceCluster_basicV2 (3576.56s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dws       3576.717s


testacc TEST=./huaweicloud/services/acceptance/dws TESTARGS='-run TestAccPublicDomainNameAssociate_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/dws -v -run TestAccPublicDomainNameAssociate_basic -timeout 360m -parallel 4
=== RUN   TestAccPublicDomainNameAssociate_basic
=== PAUSE TestAccPublicDomainNameAssociate_basic
=== CONT  TestAccPublicDomainNameAssociate_basic
--- PASS: TestAccPublicDomainNameAssociate_basic (76.04s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dws       76.309s
```

* [x] Documentation updated.
* [x] Schema updated.
* [x] CheckDeleted.

  - **a. During query operation (Read Context)**
    aa. Resource not found
   image
    
    ab. Related resources (parent resources) not found

    Cluster ID does not exist (Not created).
      image
       
    Cluster ID does not exist (Has been deleted).
    image

  - **b. During delete/disassociate/unbind operation (Delete Context)**
    ba. Resource not found
   image

    bb. Related resources (parent resources) not found
    Cluster ID does not exist.
    image

### CODE REVIEW

File: docs/resources/dws_public_domain_associate.md [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+page_title: "HuaweiCloud: huaweicloud_dws_logical_cluster"
+description: |- 
+  Use this resource to bind public domain name to DWS cluster within HuaweiCloud.
+---
+# huaweicloud_dws_public_domain_associate
+
+Use this resource to bind public domain name to DWS cluster within HuaweiCloud.
+
+-> Before using this resource, make sure that the DWS cluster has been bound to an EIP.
+   However, the EIP must be bound after the DWS cluster is created.
```
Comment (Lance52259) [inline docs/resources/dws_public_domain_associate.md]:
```suggestion
-> Before using this resource, make sure that the EIP has been bound to the DWS cluster.
```

File: docs/resources/dws_public_domain_associate.md [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+The following arguments are supported:
+
+* `region` - (Optional, String, ForceNew) Specifies the region in which to create the resource.
+  If omitted, the provider-level region will be used.
+  Changing this creates a new resource.
+
+* `cluster_id` - (Required, String, ForceNew) Specifies the DWS cluster ID to which the public domain name to be
+  associated belongs. Changing this creates a new resource.
+
+* `domain_name` - (Required, String) Specifies the domain name of the public.
```
Comment (Lance52259) [inline docs/resources/dws_public_domain_associate.md]:
```suggestion
* `domain_name` - (Required, String) Specifies the public domain name.  
```

File: docs/resources/dws_public_domain_associate.md [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+* `domain_name` - (Required, String) Specifies the domain name of the public.  
+  The valid length is limited from `4` to `64`, only English letters, digits and hyphens (-) are
+  allowed.  
+  The name must start with a English letter.
+
+* `ttl` - (Optional, Int) Specifies cache period of the SOA record set, in seconds.  
+  The valid value ranges from `300` to `2,147,483,647`. The default value is `300`.
+
+  -> If you want to modify the `ttl` value, the `domain_name` value must also be modified synchronously.
```
Comment (Lance52259) [inline docs/resources/dws_public_domain_associate.md]:
```suggestion
  -> If you want to modify the `ttl` value, the `domain_name` value must also be modified at the same time.
```

File: docs/resources/dws_public_domain_associate.md [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+
+## Attribute Reference
+
+In addition to all arguments above, the following attributes are exported:
+
+* `id` - The resource ID, also `domain_name`
```
Comment (Lance52259) [inline docs/resources/dws_public_domain_associate.md]:
```suggestion
* `id` - The resource ID, also `domain_name`.
```

File: docs/resources/dws_public_domain_associate.md [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+  If omitted, the provider-level region will be used.
+  Changing this creates a new resource.
+
+* `cluster_id` - (Required, String, ForceNew) Specifies the DWS cluster ID to which the public domain name to be
+  associated belongs. Changing this creates a new resource.
+
+* `domain_name` - (Required, String) Specifies the public domain name.  
+  The valid length is limited from `4` to `64`, only English letters, digits and hyphens (-) are
+  allowed.  
+  The name must start with a English letter.
```
Comment (ruwenqiang123) [inline docs/resources/dws_public_domain_associate.md]:
```suggestion
  The valid length is limited from `4` to `64`, only English letters, digits and hyphens (-) are allowed.  
  The name must start with an English letter.
```

File: docs/resources/dws_public_domain_associate.md [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+
+Note that the imported state may not be identical to your resource definition, due to some attributes missing from the
+API response, security or some other reason.
+The missing attributes include: `ttl`. It is generally recommended running `terraform plan` after importing the resource.
+You can then decide if changes should be applied to the instance, or the resource definition should be updated to
+align with the instance. Also you can ignore changes as below.
+
+```hcl
+resource "huaweicloud_dws_public_domain_associate" "test" {
+    ...
```
Comment (Lance52259) [inline docs/resources/dws_public_domain_associate.md]:
```suggestion
  ...
```

File: huaweicloud/services/acceptance/dws/resource_huaweicloud_dws_public_domain_associate_test.go [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+func testPublicDomainNameAssociateImportState(rName string) resource.ImportStateIdFunc {
+	return func(s *terraform.State) (string, error) {
+		rs, ok := s.RootModule().Resources[rName]
+		if !ok {
+			return "", fmt.Errorf("resource (%s) not found: %s", rName, rs)
+		}
+		clusterId := rs.Primary.Attributes["cluster_id"]
+		domainName := rs.Primary.ID
+		if clusterId == "" || domainName == "" {
+			return "", fmt.Errorf("invalid format specified for import ID, want '<cluster_id>/<domain_name>', but '%s/%s'",
```
Comment (Lance52259) [inline huaweicloud/services/acceptance/dws/resource_huaweicloud_dws_public_domain_associate_test.go]:
```suggestion
			return "", fmt.Errorf("invalid format specified for import ID, want '<cluster_id>/<domain_name>', but got '%s/%s'",
```

File: huaweicloud/services/dws/resource_huaweicloud_dws_cluster.go [hunk 8]
```diff
@@ -735,38 +708,42 @@ func clusterWaitingForAvailable(ctx context.Context, d *schema.ResourceData, met
 	return err
 }
 
+// GetClusterInfoByClusterId is a method is used to query DWS cluster details.
```
Comment (Lance52259) [inline huaweicloud/services/dws/resource_huaweicloud_dws_cluster.go]:
```suggestion
// GetClusterInfoByClusterId is a method that used to query DWS cluster detail.
```

File: huaweicloud/services/dws/resource_huaweicloud_dws_public_doamin_associate.go [hunk 9]
```diff
@@ -0,0 +1,222 @@
...
+	}
+
+	return nil
+}
+
+func resourcePublicDomainAssociateImportState(_ context.Context, d *schema.ResourceData, _ interface{}) ([]*schema.ResourceData, error) {
+	importId := d.Id()
+	parts := strings.Split(importId, "/")
+	if len(parts) != 2 {
+		return nil, fmt.Errorf("invalid format specified for import ID, want '<cluster_id>/<domain_name>', but '%s'",
```
Comment (Lance52259) [inline huaweicloud/services/dws/resource_huaweicloud_dws_public_doamin_associate.go]: ditto

File: docs/resources/dws_public_domain_associate.md (hunk 6 above)
Comment (wuzhuanhong) [reply docs/resources/dws_public_domain_associate.md]: fixed

File: huaweicloud/services/acceptance/dws/resource_huaweicloud_dws_public_domain_associate_test.go (hunk 7 above)
Comment (wuzhuanhong) [reply huaweicloud/services/acceptance/dws/resource_huaweicloud_dws_public_domain_associate_test.go]: fixed

File: huaweicloud/services/dws/resource_huaweicloud_dws_cluster.go (hunk 8 above)
Comment (wuzhuanhong) [reply huaweicloud/services/dws/resource_huaweicloud_dws_cluster.go]: fixed

File: huaweicloud/services/dws/resource_huaweicloud_dws_public_doamin_associate.go (hunk 9 above)
Comment (wuzhuanhong) [reply huaweicloud/services/dws/resource_huaweicloud_dws_public_doamin_associate.go]: fixed

**Pull request merged**